examples/bri-3/src/bri/transactions/agents/transactionStorage.agent.ts

```diff
@@ -0,0 +1,102 @@
+import { Injectable, NotFoundException } from '@nestjs/common';
+import { PrismaService } from '../../../../prisma/prisma.service';
+import { NOT_FOUND_ERR_MESSAGE } from '..//api/err.messages';
+import { Transaction } from '../models/transaction';
+
+@Injectable()
+export class TransactionStorageAgent extends PrismaService {
+  async getTransactionById(id: string): Promise<Transaction> {
+    const transactionModel = await this.transaction.findUnique({
+      where: { id: id },
+    });
+
+    if (!transactionModel) {
+      throw new NotFoundException(NOT_FOUND_ERR_MESSAGE);
+    }
+
+    return new Transaction(
+      transactionModel.id,
+      transactionModel.nonce,
+      transactionModel.workflowInstanceId,
+      transactionModel.workstepInstanceId,
+      null, // TODO: transactionModel.from once BpiAccount in the prisma schema,
+      null, // TODO: transactionModel.to once BpiAccount in the prisma schema,
+      transactionModel.payload,
+      transactionModel.signature,
+      transactionModel.status,
+    );
+  }
+
+  async getAllTransactions(): Promise<Transaction[]> {
```

Comment:
@ognjenkurtic I find that function dangerous since it is not constrained at all. What is the scope of the query? For one or more work steps? one or more workflows? one or more workgroups? a BPISubject? A BPIAccount?

Until we have understood how to properly define scope and pass that into the query we should either take it out or have a scope object as the input parameter we define later. Thoughts?

Reply:
I actually removed getAll from the API for the exact same reason but forgot to remove from storage agent. Now removed. Will add back once we have a better understanding of the scope, as  you suggested.